tiledb/sm/subarray/range_subset.cc

```diff
@@ -131,6 +132,16 @@ RangeSetAndSuperset::RangeSetAndSuperset(
     ranges_.emplace_back(superset);
 }
 
+RangeSetAndSuperset::RangeSetAndSuperset(
+    Datatype datatype,
+    const Range& superset,
+    std::vector<Range>&& subset,
+    bool coalesce_ranges)
+    : impl_(range_subset_internals(datatype, superset, coalesce_ranges))
+    , is_implicitly_initialized_(false)
```

Comment:
I don't think this line is correct. Implicit initialization is used IIRC to mean "the whole range", which is the case if the vector of subset ranges is empty.